### ISSUE TITLE
Leading spaces of texts are lost when pulling

### ISSUE THREAD

**Issue by sandraros** [larshp/abapGit#2571]
My application has a class pool with a text pool whose some symbols start with a leading space.

When pushing it to the Git repository, no issue, the XML is correctly serialized, the leading space is kept:

    <item>
     <ID>I</ID>
     <KEY>013</KEY>
     <ENTRY>   and</ENTRY>
     <LENGTH>6</LENGTH>
    </item>

When pulling from another system, the deserialization removes the leading space, consequently a difference is always shown for this object in the abapGit repository main view (and the application will behave differently too).

![image](https://user-images.githubusercontent.com/34005250/54887464-ae1ff600-4e93-11e9-9dc7-fad7fac0d53d.png)

Reason: the method PARSE of ZCL_ABAPGIT_XML does a `li_parser->set_normalizing( iv_normalize )` whose parameter has the default value "true", which strips leading spaces. 

Possible solution: I tried `li_parser->add_strip_space_element( )` instead of using the normalization, which seems to work in my case, but I fear of side effects throughout all abapGit features. Note that `li_parser->set_normalizing( abap_false )` is not sufficient alone, add_strip_... is the shortest way to solve.

Of course, putting a space at the beginning of a text can be considered a bad practice, but I feel that abapGit should not impact the way developers work. But if it's too dangerous to change abapGit, then it will just be a usage limit which needs to refactor the application (are there other ones, by the way?)

**Comment by flaiker** [comment]
In my opinion this is a bug in abapGit and should be fixed. Should also be possible to have this checked in [abapGit/CI](https://github.com/abapGit/CI).

**Comment by sandraros** [comment]
Another effect of the normalization is to remove the trailing spaces. Here is one issue with another type of object (ENHO of type clasenh, where a standalone type is added to a standard class):
![abapgit enho clif standalone type](https://user-images.githubusercontent.com/34005250/54947810-78364c80-4f3b-11e9-9d65-ed76a2410804.png)
I could check that `add_strip_space_element` also solves this issue. I'm unable to evaluate the impact of changing the method PARSE of ZCL_ABAPGIT_XML. I guess that `add_strip_space_element` could lead to issues too.

**Comment by sandraros** [comment]
Another effect of the normalization is to remove the trailing spaces. Here is one issue with another type of object (ENHO of type clasenh, where a standalone type is added to a standard class):
![abapgit enho clif standalone type](https://user-images.githubusercontent.com/34005250/54947810-78364c80-4f3b-11e9-9d65-ed76a2410804.png)
I could check that `add_strip_space_element` also solves this issue.

I'm unable to evaluate if the change of the method PARSE of `ZCL_ABAPGIT_XML` can lead to side effects elsewhere. To avoid side effects, one simple solution is to derive `ZCL_ABAPGIT_XML_INPUT` into `ZCL_ABAPGIT_XML_INPUT_STRIP` to do a strip instead of a normalize, and use this latter class wherever the "strip" is needed & tested.

**Comment by flaiker** [comment]
Added some additional text elements in abapGit-tests/CLAS_textpool@a34e23e.
I was unable to maintain any trailing spaces (by directly inputting them or using the sync feature) so I assume that's just not possible.
I tried to figure out when `li_parser->set_normalizing` was added but git blame on GitHub breaks somewhere 3 years ago (7618cada582de26569a0fe918295e3795d277a7c), when there was only one huge file.

I'd suggest to first try and fix it in a general way and run the unit tests + abapGit/CI. That should at least indicate if everything is broken afterwards.
It also seems like trailing and leading spaces as well as other special characters are not included in the XML test classes.
